### PR TITLE
Feature/safe owner cleanup

### PR DESCRIPTION
**Pull request by kopy-kat**

### CODE REVIEW

File: src/core/SafeOp.sol [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                safeOpStructHash := keccak256(encodedSafeOp, 448)
+            }
+
+            operationData =
+                abi.encodePacked(bytes1(0x19), bytes1(0x01), domainSeparator(), safeOpStructHash);
+        }
+    }
+
+    function domainSeparator() public view returns (bytes32) {
+        return keccak256(abi.encode(DOMAIN_SEPARATOR_TYPEHASH, block.chainid, this));
```
Comment (kopy-kat) [inline src/core/SafeOp.sol:98]: the launchpad is delegatecalled in the normal flow so `this` will actually be `userOp.sender`; should we change this?

File: src/Safe7579Launchpad.sol [hunk 2]
```diff
@@ -286,13 +277,14 @@ contract Safe7579Launchpad is IAccount, SafeStorage, ISafeOp {
...
-
+        (
+            bytes memory operationData,
+            uint48 validAfter,
+            uint48 validUntil,
+            bytes calldata signatures
+        ) = _getSafeOp(userOp, SUPPORTED_ENTRYPOINT);
         bytes32 _hash = keccak256(operationData);
 
         InitData memory safeSetupCallData = abi.decode(userOp.callData[4:], (InitData));
```
Comment (kopy-kat) [inline src/Safe7579Launchpad.sol:290]: technically theres an issue in the next line with using `safeSetupCallData.threshold`: 
1. there is 1 owner and threshold of 1
2. a user supplies two sigs, one of which is from the owner
3. if the wrong sig comes first, then the validation will fail even if there is a correct sig that comes second

probably not an issue but wanted to mention this

File: src/Safe7579Launchpad.sol [hunk 3]
```diff
@@ -227,48 +230,36 @@ contract Safe7579Launchpad is IAccount, SafeStorage, ISafeOp {
...
+
+            if (validatorModule == validator) userOpValidatorInstalled = true;
+        }
+
+        // Ensure that the validator module selected in the userOp was
+        // part of the validators in InitData
+        if (userOpValidatorInstalled) {
+            // validate userOp with selected validation module.
+            validationData = IValidator(validator).validateUserOp(userOp, userOpHash);
+        } else {
```
Comment (kopy-kat) [inline src/Safe7579Launchpad.sol:252]: note: this is now same behavior as safe7579 where it falls back to safe not just on zero nonce key but on any nonce key that is not an installed validator - is this still what we want?

Comment (zeroknots) [reply src/Safe7579Launchpad.sol:253]: I think for launchpad zero nonce key is sufficient?!

File: src/Safe7579Launchpad.sol (hunk 2 above)
Comment (zeroknots) [reply src/Safe7579Launchpad.sol:291]: good should. should we require sorted instead?

File: src/core/SupportViewer.sol [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+import {
+    MODULE_TYPE_VALIDATOR,
+    MODULE_TYPE_HOOK,
+    MODULE_TYPE_EXECUTOR,
+    MODULE_TYPE_FALLBACK
+} from "erc7579/interfaces/IERC7579Module.sol";
+
+contract SupportViewer {
+    function accountId() external view returns (string memory accountImplementationId) {
+        return "safe7579.v1.0.0";
```
Comment (zeroknots) [inline src/core/SupportViewer.sol:24]: maybe we should put "launchpad" in there?

File: src/Safe7579.sol [hunk 5]
```diff
@@ -50,7 +51,7 @@ uint256 constant MULTITYPE_MODULE = 0;
  * event emissions to be done via the SafeProxy as msg.sender using Safe's
  * "executeTransactionFromModule" features.
  */
-contract Safe7579 is ISafe7579, ISafeOp, AccessControl, Initializer {
+contract Safe7579 is ISafe7579, SafeOp, SupportViewer, AccessControl, Initializer {
```
Comment (zeroknots) [inline src/Safe7579.sol:54]: ✅

File: src/Safe7579Launchpad.sol [hunk 6]
```diff
@@ -123,17 +123,21 @@ contract Safe7579Launchpad is IAccount, SafeStorage, ISafeOp {
     function addSafe7579(
         address safe7579,
         ModuleInit[] calldata validators,
+        ModuleInit[] calldata executors,
+        ModuleInit[] calldata fallbacks,
+        ModuleInit[] calldata hooks,
         address[] calldata attesters,
         uint8 threshold
     )
         external
```
Comment (zeroknots) [inline src/Safe7579Launchpad.sol:132]: should we add `onlyDelegatecall` here?

**Pull request merged**